asserts/account_key.go

```diff
@@ -216,14 +234,77 @@ func assembleAccountKey(assert assertionBase) (Assertion, error) {
 		return nil, err
 	}
 
+	var matchers []attrMatcher
+	if cs, ok := assert.headers["constraints"]; ok {
+		matchers, err = checkAKConstraints(cs)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	// ignore extra headers for future compatibility
 	return &AccountKey{
-		assertionBase: assert,
-		sinceUntil:    *sinceUntil,
-		pubKey:        pubk,
+		assertionBase:      assert,
+		sinceUntil:         *sinceUntil,
+		constraintMatchers: matchers,
+		pubKey:             pubk,
 	}, nil
 }
 
+func checkAKConstraints(cs interface{}) ([]attrMatcher, error) {
+	csmaps, ok := cs.([]interface{})
+	if !ok {
+		return nil, fmt.Errorf("assertions constraints must be a list of maps")
+	}
+	if len(csmaps) == 0 {
+		// there is no syntax producing this scenario but be robust
+		return nil, fmt.Errorf("assertions constraints cannot be empty")
+	}
+	matchers := make([]attrMatcher, 0, len(csmaps))
+	for _, csmap := range csmaps {
+		m, ok := csmap.(map[string]interface{})
+		if !ok {
+			return nil, fmt.Errorf("assertions constraints must be a list of maps")
+		}
+		hm, err := checkMapWhat(m, "headers", "constraint")
+		if err != nil {
+			return nil, err
+		}
+		if hm == nil {
+			return nil, fmt.Errorf(`"headers" constraint mandatory in asserions constraints`)
+		}
+		t, ok := hm["type"]
+		if !ok {
+			return nil, fmt.Errorf("type header constraint mandatory in asserions constraints")
+		}
+		tstr, ok := t.(string)
+		if !ok {
+			return nil, fmt.Errorf("type header constraint must be a string")
+		}
```

Comment:
`checkNotEmptyString`?

Reply:
the issue doing that would be less specific errors